pkg/rid/store/cockroach/identification_service_area.go

```diff
@@ -222,3 +222,26 @@ func (c *isaRepo) SearchISAs(ctx context.Context, cells s2.CellUnion, earliest *
 
 	return c.process(ctx, isasInCellsQuery, earliest, latest, pq.Int64Array(cids))
 }
+
+// ListExpiredISAs lists all expired ISAs based on writer.
+// Records expire if current time is <expiredDurationInMin> minutes more than records' endTime.
+func (c *isaRepo) ListExpiredISAs(ctx context.Context, writer *string) ([]*ridmodels.IdentificationServiceArea, error) {
+	writerQuery := "NULL"
+	if writer != nil {
```

Comment:
With this implementation, if we want to garbage-collect everything without a writer in a robust way, we'll need to call ListExpiredISAs(ctx, "") and ListExpiredISAs(ctx, nil) -- is there ever a case in which we would want to make those calls separately, or distinguish between entries with "" writer and null writer?  If not, let's accept a standard string argument and query for both "" and null when the empty string is supplied.